### ISSUE TITLE
Input does not pass WCAG 2.1

### ISSUE THREAD

**Issue by DanielWalker99** [NHSDigital/nhsuk-react-components#134]
Hi
Currently, the input component causes an alert when checking against WCAG 2.1 as shown in an example from the storybook below 
![image](https://user-images.githubusercontent.com/72249213/222129603-46549d0a-38b5-4a6c-87ae-46998a51a9bd.png)

Viewing https://www.w3.org/WAI/ARIA/apg/practices/names-and-descriptions/ in the section named **Naming Form Controls with the Label Element** it appears that the input should be wrapped in a label in order to allow for assistive technology to link a label with its corresponding input.

**Comment by edwardhorsford** [comment]
The NHS design system reference implementation (which I think this port should aim to replicate) does this:
```html
<div class="nhsuk-form-group">
  <label class="nhsuk-label" for="example">
    Full name
  </label>
  <input class="nhsuk-input" id="example" name="example" type="text">
</div>
```

Checking the rendered markup in the current storybook I get this:
```html
<div class="nhsuk-form-group">
  <label class="nhsuk-label" id="input-example--label" for="input-example">
  National Insurance Number</label>
  <input class="nhsuk-input" aria-labelledby="input-example--label" name="test-name" id="input-example" type="text">
</div>
```

Wrapping an input in a label is one way of linking them, but using `for="foo"` is another. In this case I would say this library should be outputting the same markup as the design system - which has been extensively accessibility tested. This library is also using `aria-labelledby` which I think should probably be removed.

**Comment by edwardhorsford** [comment]
Closing as I don't think this is an issue. Please comment if you dissagree.
